api/pkg/apis/v1alpha1/providers/states/k8s/k8s.go

```diff
@@ -243,21 +248,24 @@ func (s *K8sStateProvider) Upsert(ctx context.Context, entry states.UpsertReques
 			}
 		}
 		if v, ok := dict["status"]; ok {
-			status := &unstructured.Unstructured{
-				Object: map[string]interface{}{
-					"apiVersion": group + "/" + version,
-					"kind":       "Status",
-					"metadata": map[string]interface{}{
-						"name": entry.Value.ID,
+			statusMap := v.(map[string]interface{})
+			if p, ok := statusMap["properties"]; ok {
+				status := &unstructured.Unstructured{
+					Object: map[string]interface{}{
+						"apiVersion": group + "/" + version,
+						"kind":       "Status",
+						"metadata": map[string]interface{}{
+							"name": entry.Value.ID,
+						},
+						"properties": p.(map[string]interface{}),
```

Comment:
Why we save this in "properties" field? Seems this is copied from dict["status"].